twitter-clone/src/AuthContext.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { createContext, useState, useContext } from 'react';
+
+export interface User {
+  username: string;
+  id: string;
+  name: string;
+  email: string;
+}
+
+interface AuthContextType {
+  user: User | null;
+  logIn: (userData: User) => void;
+  logOut: () => void;
+}
+
+const defaultAuthContext = {
+  user: null,
+  logIn: () => {},
+  logOut: () => {},
+};
+
+export const useAuth = (): AuthContextType => {
+  const context = useContext(AuthContext);
+  if (context === undefined) {
+    throw new Error('useAuth must be used within an AuthProvider');
+  }
+  return context;
+};
+
+export const AuthContext = createContext<AuthContextType>(defaultAuthContext);
+export const AuthProvider: React.FC = ({ children }) => {
```

Comment:
Spaces between entities and js docs is good approach that increase code redability.